src/main/java/net/dv8tion/jda/api/entities/MessageMentions.java

```diff
@@ -0,0 +1,220 @@
+package net.dv8tion.jda.api.entities;
+
+import gnu.trove.set.TLongSet;
+
+import java.util.*;
+
+import java.util.stream.Collectors;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.requests.RestAction;
+import net.dv8tion.jda.api.utils.MiscUtil;
+import net.dv8tion.jda.internal.JDAImpl;
+import net.dv8tion.jda.internal.entities.EmoteImpl;
+import net.dv8tion.jda.internal.utils.Checks;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.function.Function;
+import java.util.regex.Matcher;
+
+public class MessageMentions
+{
+    private final String content;
+    private final boolean mentionsEveryone;
+    private final TLongSet userIds;
+    private final TLongSet roleIds;
+    private final JDAImpl jda;
+    private final Guild guild;
+
+    private final List<User> users;
+    private final List<Member> members;
+    private final List<Role> roles;
+    private final List<TextChannel> textChannels;
+    private final List<VoiceChannel> voiceChannels;
+
+    private final List<Emote> emotes;
+    private final List<String> invites;
+    private final List<MessageSticker> stickers;
+
+    public MessageMentions(
+            String content, boolean mentionsEveryone, TLongSet userIds, TLongSet roleIds, List<MessageSticker> stickers,
+            Guild guild, JDAImpl jda
+    ) {
+        this.content = content;
+        this.mentionsEveryone = mentionsEveryone;
+        this.userIds = userIds;
+        this.roleIds = roleIds;
+        this.jda = jda;
+        this.guild = guild;
+
+        this.users = Collections.unmodifiableList(processMentions(Message.MentionType.USER, new ArrayList<>(), true, this::matchUser));
```

Comment:
You wrap a lot of collections returned by `processMentions()` with a `Collections.unmodifiableList()` call.
Maybe change the methods returned collection.
```suggestion
        this.users = processMentions(Message.MentionType.USER, new ArrayList<>(), true, this::matchUser);
```